nifi-extension-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -90,77 +93,111 @@ public class GetWorkdayReport extends AbstractProcessor {
-        .description("The username provided for authentication of Workday requests. Encoded using Base64 for HTTP Basic Authentication as described in RFC 7617.")
-        .required(true)
-        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x39\\x3b-\\x7e\\x80-\\xff]+$")))
-        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
-        .build();
+            .name("Workday Username")
+            .displayName("Workday Username")
+            .description("The username provided for authentication of Workday requests. Encoded using Base64 for HTTP Basic Authentication as described in RFC 7617.")
+            .dependsOn(AUTH_TYPE, BASIC_AUTH_TYPE)
+            .required(true)
+            .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x39\\x3b-\\x7e\\x80-\\xff]+$")))
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .build();
 
     protected static final PropertyDescriptor WORKDAY_PASSWORD = new PropertyDescriptor.Builder()
-        .name("Workday Password")
-        .displayName("Workday Password")
-        .description("The password provided for authentication of Workday requests. Encoded using Base64 for HTTP Basic Authentication as described in RFC 7617.")
-        .required(true)
-        .sensitive(true)
-        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x7e\\x80-\\xff]+$")))
-        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
-        .build();
+            .name("Workday Password")
+            .displayName("Workday Password")
+            .description("The password provided for authentication of Workday requests. Encoded using Base64 for HTTP Basic Authentication as described in RFC 7617.")
+            .dependsOn(AUTH_TYPE, BASIC_AUTH_TYPE)
+            .required(true)
+            .sensitive(true)
+            .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x7e\\x80-\\xff]+$")))
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .build();
 
     protected static final PropertyDescriptor WEB_CLIENT_SERVICE = new PropertyDescriptor.Builder()
-        .name("Web Client Service Provider")
-        .description("Web client which is used to communicate with the Workday API.")
-        .required(true)
-        .identifiesControllerService(WebClientServiceProvider.class)
-        .build();
+            .name("Web Client Service Provider")
+            .description("Web client which is used to communicate with the Workday API.")
+            .required(true)
+            .identifiesControllerService(WebClientServiceProvider.class)
+            .build();
+
+    public static final PropertyDescriptor OAUTH2_ACCESS_TOKEN_PROVIDER =
+            new PropertyDescriptor.Builder()
+                    .name("Access Token Provider")
+                    .displayName("OAuth2 Access Token Provider")
```

Comment:
The `displayName` should be removed as noted on other properties.
```suggestion
                    .name("Access Token Provider")
```